app/vep/utils/nextflow.py

```
@@ -0,0 +1,21 @@
+import requests
+import json
+from core.config import NF_TOKEN, NF_PIPELINE_URL, NF_COMPUTE_ENV_ID, NF_WORK_DIR, SEQERA_WORKFLOW_LAUNCH_URL
+from models.pipeline_model import PipelineParams
+
+def launch_workflow(pipeline_params: PipelineParams):
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {NF_TOKEN}"
+    }
+
+    response = requests.post(SEQERA_WORKFLOW_LAUNCH_URL, headers=headers, json=pipeline_params)
+
+    # Check if successful
+    if response.status_code == 200:
+        return response.json()
+    else:
+        print(f"Failed to launch workflow. Status code: {response.status_code}")
+        return response.text
```

Shouldn't the function raise some kind of error if the request to the platform failed? If I am reading the code correctly, all errors will be quietly swallowed.

---

As I said earlier, error handling is not required in this ticket, but since you mentioned I will add it.

---

> As I said earlier, error handling is not required in this ticket, but since you mentioned I will add it.

Sorry, after seeing the new commit, I realised that there must have been a bit of a misunderstanding. I thought the `launch_workflow` function would itself be called by some other function (the actual request handler); and the only responsibility of `launch_workflow` would be to raise an exception in case of a failed request to Seqera, which would be ultimately handled up the chain, in the request handler. I didn't expect that the `launch_workflow` function should actually return a Response object. I may well be wrong.

---

hmm it makes sense for the launch_workflow to just raise an error and the caller handles it accordingly. Lets discuss in standup